### PR TITLE
the gem is added the state machine is created

### PR DESCRIPTION
**Pull request by NadiaArroyo25**
In this branch there is also the language change,  the time zone configuration and the task

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -233,6 +233,13 @@ GEM
       actionpack (>= 5.2)
       activesupport (>= 5.2)
       sprockets (>= 3.0.0)
+    state_machines (0.6.0)
```
Comment (NadiaArroyo25) [inline Gemfile.lock:236]: These changes were automatic

File: Gemfile.lock [hunk 2]
```diff
@@ -233,6 +233,13 @@ GEM
...
+    state_machines-activemodel (0.9.0)
+      activemodel (>= 6.0)
+      state_machines (>= 0.6.0)
+    state_machines-activerecord (0.9.0)
+      activerecord (>= 6.0)
+      state_machines-activemodel (>= 0.9.0)
```
Comment (NadiaArroyo25) [inline Gemfile.lock:242]: These changes were automatic

File: Gemfile.lock [hunk 3]
```diff
@@ -281,6 +288,8 @@ DEPENDENCIES
   selenium-webdriver
   simple_calendar
   sprockets-rails
+  state_machines
+  state_machines-activerecord
```
Comment (NadiaArroyo25) [inline Gemfile.lock:292]: These changes were automatic

File: Gemfile [hunk 4]
```diff
@@ -23,12 +23,13 @@ gem "stimulus-rails"
 
 # Build JSON APIs with ease [https://github.com/rails/jbuilder]
 gem "jbuilder"
-gem "pry"  
+gem "pry"
```
Comment (Ligator) [inline Gemfile]: Move this gem to inside the developer and test environments.

File: app/models/user.rb [hunk 5]
```diff
@@ -4,4 +4,7 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
   has_many :medical_records
+  validates :password, presence: true, length: { minimum: 8 }
```
Comment (Ligator) [inline app/models/user.rb]: The password is handled by the devise gem, not as a regular attribute.
The reason is that the password is not stored in plain text, so this validation is not actually counting the password, instead, it is counting a string that represents the encrypted password and that is always more than 8 characters.
So, please look for the devise config file in your project, you should see there the line to setup the password length.

**Comment by Ligator** [comment]
Merge main into this branch.

File: Gemfile (hunk 4 above)
Comment (NadiaArroyo25) [reply Gemfile]: vale

File: config/initializers/devise.rb [hunk 6]
```diff
@@ -178,7 +178,7 @@
 
   # ==> Configuration for :validatable
   # Range for password length.
-  config.password_length = 6..128
+  config.password_length = 8..128
```
Comment (Ligator) [inline config/initializers/devise.rb:181]: 👏

**Review by Ligator — APPROVED** [review]

**Pull request merged**